zookeeper-server/src/main/java/org/apache/zookeeper/server/quorum/QuorumCnxManager.java

```diff
@@ -1390,7 +1397,7 @@ public void run() {
                      * Allocates a new ByteBuffer to receive the message
                      */
                     final byte[] msgArray = new byte[length];
-                    din.readFully(msgArray, 0, length);
+                    din.read(msgArray);
                     addToRecvQueue(new Message(ByteBuffer.wrap(msgArray), sid));
```

Comment:
Did you mean `din.readFully(msgArray)`?  A bare `read` only "reads some number of bytes," and there is no guarantee we don't run into a fragmented packet or something else at this point, is there?